### PR TITLE
chore : Upgrade keycloak to 19.0.0

### LINKED ISSUE #1009: Upgrade to Keycloak 19
Similar to what was done here by @yelhouti :
- https://github.com/jhipster/generator-jhipster/pull/18441

### PR DESCRIPTION
**Pull request by Rajpratik71**
Closes #1009 

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

### CODE REVIEW

File: src/main/resources/generator/server/springboot/mvc/security/oauth2/docker/jhipster-realm.json.mustache [hunk 1]
```diff
@@ -2250,5 +2250,63 @@
   },
   "clientPolicies": {
     "policies": []
-  }
+  },
+  "users": [
```
Comment (DamnClin) [inline src/main/resources/generator/server/springboot/mvc/security/oauth2/docker/jhipster-realm.json.mustache:2254]: Is this merge really mandatory? It's not possible to keep files separated with this version?

File: src/test/resources/generator/dependencies/Dockerfile [hunk 2]
```diff
@@ -2,7 +2,7 @@ FROM sonarqube:9.3.0-community
 FROM consul:1.11.4
 FROM jhipster/consul-config-loader:v0.4.1
 FROM jhipster/jhipster-registry:v7.3.0
-FROM jboss/keycloak:16.1.1
+FROM quay.io/keycloak/keycloak:19.0.0
```
Comment (DamnClin) [inline src/test/resources/generator/dependencies/Dockerfile:5]: As I'm not into this at all I found it strange to rely on what's looks to be a third party to provide a keycloak image, can you give some explanations on that?

**Comment by DamnClin** [comment]
Looks like you have a formatting problem. You can run: 

```
npm i
npm run prettier:format
```

(The manual formatting is only needed here since `npm i` will install a git hook to handle formatting during commits)

File: src/test/resources/generator/dependencies/Dockerfile (hunk 2 above)
Comment (Rajpratik71) [reply src/test/resources/generator/dependencies/Dockerfile:5]: This is the official image of keycloak, from long time it is being published at quay.io/keycloak/keycloak

Official email thread at https://lists.jboss.org/pipermail/keycloak-user/2019-March/017456.html

**Comment by DamnClin** [comment]
You need to update `OAuth2ModuleFactory` to use the new image name at this line: 

```
DockerImage keycloakImage = dockerImages.get("jboss/keycloak");
```

You then need to update `OAuth2ModuleFactoryTest` and change mock and expected value (this mock for test is not really great, we should work on something using the real image name and just setting the version)

File: src/main/resources/generator/server/springboot/mvc/security/oauth2/docker/jhipster-realm.json.mustache (hunk 1 above)
Comment (DamnClin) [reply src/main/resources/generator/server/springboot/mvc/security/oauth2/docker/jhipster-realm.json.mustache:2254]: I really don't like this file merge, it is not possible to keep two files here?

**Comment by DamnClin** [comment]
Ok, just wasted one hour trying to figure out how to have multiple files for the same realm at startup and that just seems to be a real pain to do (at least the current keycloak codebase don't seems to have something allowing that easily) so this is OK for me.

@pascalgrimaud ok for you (as you probably know this part way better than I am), I merge?

**Comment by DamnClin** [comment]
How, just realized that I'm not supposed to merge this since there's a Bounty on it!

**Comment by pascalgrimaud** [comment]
@DamnClin : if it works, you can merge it, no worry about the bounty as once it's submitted, the admins (Julien, Deepu, Daniel and me) will be notified and can approve it

**Comment by DamnClin** [comment]
Yep, taht works, thanks a lot @Rajpratik71, don't forget to claim your bounty :) https://www.jhipster.tech/bug-bounties/#how-to-get-the-money

**Comment by Rajpratik71** [comment]
> Yep, that works, thanks a lot @Rajpratik71, don't forget to claim your bounty :) https://www.jhipster.tech/bug-bounties/#how-to-get-the-money

Sure thanks @DamnClin @pascalgrimaud

**Comment by Rajpratik71** [comment]
Raised claim at https://opencollective.com/generator-jhipster/expenses/89411

**Pull request merged**